controllers/ndbserver_controller_helpers.go

```diff
@@ -0,0 +1,84 @@
+package controllers
+
+import (
+	"context"
+
+	ndbv1alpha1 "github.com/nutanix-cloud-native/ndb-operator/api/v1alpha1"
+	"github.com/nutanix-cloud-native/ndb-operator/common"
+	"github.com/nutanix-cloud-native/ndb-operator/common/util"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_api"
+	"github.com/nutanix-cloud-native/ndb-operator/ndb_client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Fetches all databases from the NDB API and converts them to
+// NDBServerDatabaseInfo type object (to be consumed by the NDBServer CR)
+func getNDBServerDatabasesInfo(ctx context.Context, ndbClient *ndb_client.NDBClient) (databases []ndbv1alpha1.NDBServerDatabaseInfo, err error) {
+	log := log.FromContext(ctx)
+	log.Info("Entered ndbserver_controller_helpers.getNDBServerDatabasesInfo")
```

Comment:
if we are retaining logs for such things then we should log something more meaningful (even if its an internal log)